### PR TITLE
KAFKA-17418: Fix the incorrect markdown of junit.py caused by newline character

### PR DESCRIPTION
**Pull request by xijiu**
As title, see https://github.com/apache/kafka/actions/runs/10534117374/jobs/21477070226/summary_raw

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@xijiu Could you please rebase code to trigger github CI again?

**Comment by xijiu** [comment]
> @xijiu Could you please rebase code to trigger github CI again?

@chia7712  Sure, I have rebased, PTAL

**Review by mumrah — COMMENTED** [review]
@xijiu thanks for the patch! I think we can replace newlines with `<br>` to preserve some of the formatting. Also, we should probably escape any HTML-looking things in the output to avoid similar formatting problems in the future. 

In Python there is `html.escape` for this.

Also, a small tip: you can use the PR comments as a sandbox for testing Github flavored markdown.

**Comment by xijiu** [comment]
> @xijiu thanks for the patch! I think we can replace newlines with `<br>` to preserve some of the formatting. Also, we should probably escape any HTML-looking things in the output to avoid similar formatting problems in the future.
> 
> In Python there is `html.escape` for this.
> 
> Also, a small tip: you can use the PR comments as a sandbox for testing Github flavored markdown.

@mumrah   
Hi David, thanks very much for the code review, and the tip is very useful.
I have fixed the code, PTAL

File: .github/scripts/junit.py [hunk 1]
```diff
@@ -95,6 +96,9 @@ def parse_report(workspace_path, report_path, fp) -> Iterable[TestSuite]:
                 test_case_failed = False
             elif elem.tag == "failure":
                 failure_message = elem.get("message")
+                if failure_message:
+                    failure_message = failure_message.replace('\n', '<br>').replace('\r', '<br>')
+                    failure_message = html.escape(failure_message)
```
Comment (mumrah) [inline .github/scripts/junit.py]: I think these should be reversed. Otherwise the `<br>` will get escaped and not actually rendered as HTML

**Review by mumrah — COMMENTED** [review]
Thanks for the updates @xijiu! One last comment

File: .github/scripts/junit.py (hunk 1 above)
Comment (xijiu) [reply .github/scripts/junit.py]: Yeah, you are right, I should adjust the order

**Comment by xijiu** [comment]
@mumrah  

Hello David, I have fixed the code,  PTAL

**Review by mumrah — APPROVED** [review]
LGTM. Thanks @xijiu 👍

**Pull request merged**